pyhealth/datasets/eicu.py

```diff
@@ -563,11 +567,148 @@ def physicalExam_unit(v_info):
+        """Helper function which parses diagnosis string table, which is the diagnosisString
+        column under the diagnosis table.
+
+        Will be called in `self.parse_tables()`.
+
+        Docs:
+            - diagnosis: https://eicu-crd.mit.edu/eicutables/diagnosis/
+
+        Args:
+            patients: a dict of `Patient` objects indexed by patient_id.
+
+        Returns:
+            The updated patients dict.
+        """
+        table = "diagnosis"
+        # read table
+        df = pd.read_csv(
+            os.path.join(self.root, f"{table}.csv"),
+            dtype={"patientunitstayid": str, "diagnosisstring": str},
+        )
+        # drop rows with missing values
+        df = df.dropna(subset=["patientunitstayid", "diagnosisstring"])
+        # sort by diagnosisoffset
+        df = df.sort_values(["patientunitstayid", "diagnosisoffset"], ascending=True)
+        # add the patient id info
+        df["patient_id"] = df["patientunitstayid"].apply(
+            lambda x: self.visit_id_to_patient_id.get(x, None)
+        )
+        # add the visit encounter time info
+        df["v_encounter_time"] = df["patientunitstayid"].apply(
+            lambda x: self.visit_id_to_encounter_time.get(x, None)
+        )
+        # group by visit
+        group_df = df.groupby("patientunitstayid")
+
+        # parallel unit of diagnosisString (per visit)
+        def diagnosisString_unit(v_info):
+            v_id = v_info["patientunitstayid"].values[0]
+            patient_id = v_info["patient_id"].values[0]
+            v_encounter_time = v_info["v_encounter_time"].values[0]
+            if patient_id is None:
+                return []
+
+            events = []
+            for offset, code in zip(
+                    v_info["diagnosisoffset"], v_info["diagnosisstring"]
+            ):
+                # compute the absolute timestamp
+                timestamp = v_encounter_time + pd.Timedelta(minutes=offset)
+                event = Event(
+                    code=code,
+                    table="diagnosisString",
+                    vocabulary="eICU_DIAGNOSISSTRING",
+                    visit_id=v_id,
+                    patient_id=patient_id,
+                    timestamp=timestamp,
+                )
+                # update patients
+                events.append(event)
+            return events
+
+        # parallel apply
+        group_df = group_df.parallel_apply(lambda x: diagnosisString_unit(x))
+
+        # summarize the results
+        patients = self._add_events_to_patient_dict(patients, group_df)
+        return patients
+
 
 if __name__ == "__main__":
     dataset = eICUDataset(
         root="/srv/local/data/physionet.org/files/eicu-crd/2.0",
-        tables=["diagnosis", "medication", "lab", "treatment", "physicalExam"],
+        tables=["diagnosis", "medication", "lab", "treatment", "physicalExam", "admissionDx", "diagnosisString"],
```

Comment:
It is under the [diagnosis.csv](https://eicu-crd.mit.edu/eicutables/diagnosis/). Instead of using the ICD9Code, the paper used the "diagnosisString" column instead.

Reply:
I see, could you please merge the diagnosis and diagnosisString. It could be done by:
1. add an extra diagnosisString-value pair in the diagnosis Event https://pyhealth.readthedocs.io/en/latest/api/data/pyhealth.data.Visit.html
2. instead of using "get_code_list" in the task_func, please use "get_event_list" and then get the value from the "diagnosisString" key.